### PR TITLE
Show using custom nonce logic

### LINKED ISSUE #739: Document using custom nonces
### Feature Request

Document replacing `get_nonce` function implementation with custom one.

### PR DESCRIPTION
**Pull request by cptartur**
Closes #739 

## Introduced changes

This PR adds a example in guide showing using custom nonce logic in `Account`.

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/guide/account_and_client.rst [hunk 1]
```diff
@@ -14,6 +14,18 @@ Here is an example:
     :dedent: 4
 
 
+Custom nonce logic
+------------------
+
+By default, :ref:`Account` calls StarkNet for nonce every time a new transaction is signed or executed.
+This is okay in for most users, but in case your applications needs to pre-sing multiple transactions
+for execution, deals with high amount of transactions or for whatever reason needs to support different nonce
```
Comment (drknzz) [inline docs/guide/account_and_client.rst:22]:
```suggestion
for execution, deals with high amount of transactions or just needs to support different nonce
```

File: docs/guide/account_and_client.rst [hunk 2]
```diff
@@ -14,6 +14,18 @@ Here is an example:
...
+logic, it is possible to so with :ref:`Account. Simply overwrite the
```
Comment (drknzz) [inline docs/guide/account_and_client.rst:23]:
```suggestion
logic, it is possible to so with :ref:`Account`. Simply overwrite the
```

File: docs/guide/account_and_client.rst [hunk 3]
```diff
@@ -14,6 +14,18 @@ Here is an example:
...
+:meth:`~starknet_py.net.account.account.Account.get_nonce` with your own logic.
```
Comment (drknzz) [inline docs/guide/account_and_client.rst:24]:
```suggestion
:meth:`~starknet_py.net.account.account.Account.get_nonce` method with your own logic.
```

File: starknet_py/tests/e2e/docs/guide/test_custom_nonce.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        address=address,
+        client=client,
+        key_pair=KeyPair.from_private_key(private_key),
+        chain=StarknetChainId.TESTNET,
+    )
+    # docs: end
+
+    assert account.nonce_counter == 0
+    await account.sign_invoke_transaction(calls=Call(0x1, 0x1, []), max_fee=10000000000)
+    assert account.nonce_counter == 1
```
Comment (drknzz) [inline starknet_py/tests/e2e/docs/guide/test_custom_nonce.py:58]: Maybe this could also be included?

File: docs/guide/account_and_client.rst [hunk 5]
```diff
@@ -14,6 +14,18 @@ Here is an example:
...
+
+By default, :ref:`Account` calls StarkNet for nonce every time a new transaction is signed or executed.
+This is okay for most users, but in case your applications needs to pre-sign multiple transactions
+for execution, deals with high amount of transactions or just needs to support different nonce
+logic, it is possible to so with :ref:`Account`. Simply overwrite the
+:meth:`~starknet_py.net.account.account.Account.get_nonce` method with your own logic.
+
+.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_custom_nonce.py
+    :language: python
+    :dedent: 4
```
Comment (drknzz) [inline docs/guide/account_and_client.rst:28]: Maybe we should put it at the end of this section? I feel like it's not that common to use use it, and it is the second subsection in the whole guide.
Or we could put it somewhere in `signing` section maybe?

File: starknet_py/tests/e2e/docs/guide/test_custom_nonce.py (hunk 4 above)
Comment (cptartur) [reply starknet_py/tests/e2e/docs/guide/test_custom_nonce.py:58]: Don't think it's important to have that in guide. What this example is supposed to show is that it is possible to have custom nonces, not how to do it exactly.

File: docs/guide/account_and_client.rst (hunk 5 above)
Comment (cptartur) [reply docs/guide/account_and_client.rst:28]: We can move it to the end.

File: starknet_py/tests/e2e/docs/guide/test_custom_nonce.py (hunk 4 above)
Comment (drknzz) [reply starknet_py/tests/e2e/docs/guide/test_custom_nonce.py:58]: Right, but it shows how the custom class works in practice. But I'm fine with not having it though.

Comment (cptartur) [reply starknet_py/tests/e2e/docs/guide/test_custom_nonce.py:58]: Yeah, I don't think it matters how it works. Imo, the guide could even be showing `...` in place of the custom implementation.

File: docs/guide/account_and_client.rst [hunk 6]
```diff
@@ -39,3 +38,17 @@ You can use :class:`starknet_py.net.client_errors.ClientError` to catch errors f
...
+
+Custom nonce logic
+------------------
+
+By default, :ref:`Account` calls StarkNet for nonce every time a new transaction is signed or executed.
+This is okay for most users, but in case your applications needs to pre-sign multiple transactions
+for execution, deals with high amount of transactions or just needs to support different nonce
+logic, it is possible to so with :ref:`Account`. Simply overwrite the
+:meth:`~starknet_py.net.account.account.Account.get_nonce` method with your own logic.
+
```
Comment (drknzz) [inline docs/guide/account_and_client.rst:51]: Should we write here that the code below is just an example implementation of a nonce counter?

**Review by drknzz — APPROVED** [review]

**Pull request merged**